synapsemonitor/monitor.py

```diff
@@ -69,46 +69,63 @@ def _render_fileview(
     return viewdf
 
 
-def _find_modified_entities_fileview(syn: Synapse, syn_id: str, days: int = 1) -> list:
-    """Finds entities scoped in a fileview modified in the past N number of days
+def _find_modified_entities_fileview(
+    syn: Synapse, syn_id: str, value: int = 1, unit: str = "day"
+) -> list:
+    """Finds entities scoped in a fileview modified in the past {value} {unit}
 
     Args:
         syn: Synapse connection
         syn_id: Synapse Fileview Id
-        days: N number of days
+        value: number of time units
+        unit: time unit
 
     Returns:
         List of synapse ids
     """
     # Update the view
     # _force_update_view(syn, view_id)
+
     query = (
         f"select id from {syn_id} where "
-        f"modifiedOn > unix_timestamp(NOW() - INTERVAL {days} DAY)*1000"
+        f"modifiedOn > unix_timestamp(NOW() - INTERVAL {value} {unit})*1000"
     )
     results = syn.tableQuery(query)
     resultsdf = results.asDataFrame()
     return resultsdf["id"].tolist()
 
 
-def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> list:
-    """Determines if entity was modified in the past N number of days
+def _find_modified_entities_file(
+    syn: Synapse, syn_id: str, value: int = 1, unit: str = "day"
+) -> list:
+    """Determines if entity was modified in the past {value} {unit}.
+    Note: entity modifiedOn returns UTC time
 
     Args:
         syn: Synapse connection
         syn_id: Synapse File Id
-        days: N number of days
+        value: number of time units
+        unit: time unit
 
     Returns:
         List of synapse ids
     """
     entity = syn.get(syn_id, downloadFile=False)
-    # Entity modified on returns UTC time
-    utc_mod = datetime.strptime(entity["modifiedOn"], "%Y-%m-%dT%H:%M:%S.%fZ").replace(
-        tzinfo=tz.tzutc()
-    )
-    utc_now = datetime.now().replace(tzinfo=tz.tzutc())
-    if utc_mod > utc_now - timedelta(days=days):
+    utc_mod = datetime.strptime(entity["modifiedOn"], "%Y-%m-%dT%H:%M:%S.%fZ")
+    utc_now = datetime.utcnow()
+
+    if unit == "day":
+        td = timedelta(days=value)
+    elif unit == "hour":
+        td = timedelta(hours=value)
+    elif unit == "minute":
+        td = timedelta(minutes=value)
+    else:
+        raise ValueError(
+            f"'{unit}' is not an accepted time unit. Accepted units are 'day', 'hour', 'minute'."
+        )
```

Comment:
This error check should happen earlier

Reply:
Moved check as first operation in function 

Also, cli only allows `day` and `hour` as valid choices